ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -170,6 +171,11 @@ internal class PaywallViewModelImpl(
                 val customerInfo = purchases.awaitRestore()
                 Logger.i("Restore purchases successful: $customerInfo")
                 listener?.onRestoreCompleted(customerInfo)
+                shouldDisplayBlock?.let {
+                    if (!it(customerInfo)) {
+                        options.dismissRequest()
```

Comment:
I would add some logs here like we do when we determine if the paywall should be presented.